File: LeetCodeProblemsCMake/Problems/LeetCode3195FindTheMinimumAreaToCoverAllOnesI.cpp
//
// Created by saeed on 8/24/25.
//

#include "LeetCode3195FindTheMinimumAreaToCoverAllOnesI.h"
using namespace std;

int LeetCode3195FindTheMinimumAreaToCoverAllOnesI::minimumArea(vector<vector<int> > &grid) {
    int height=static_cast<int>(grid.size());
    int width= static_cast<int>(grid[0].size());
    int maxHeight=-1;
    int maxWidth=-1;
    int minHeight=height;
    int minWidth=width;
    for (int i=0;i<height;i++) {
        for (int j=0;j<width;j++) {
            if (grid[i][j]==0) {
                continue;
            }
            maxHeight=max(maxHeight,i);
            maxWidth=max(maxWidth,j);
            minHeight=min(minHeight,i);
            minWidth=min(minWidth,j);
        }
    }
    return (maxHeight-minHeight+1)*(maxWidth-minWidth+1);
}

File: LeetCodeProblemsGo/LeetCode3195FindTheMinimumAreaToCoverAllOnesI/LeetCode3195FindTheMinimumAreaToCoverAllOnesI.go
package LeetCode3195FindTheMinimumAreaToCoverAllOnesI

func minimumArea(grid [][]int) int {
	heigth, width := len(grid), len(grid[0])
	minHeigth, maxHeight, minWidth, maxWidth := heigth, -1, width, -1
	for i := 0; i < heigth; i++ {
		for j := 0; j < width; j++ {
			if grid[i][j] == 0 {
				continue
			}
			if i < minHeigth {
				minHeigth = i
			}
			if i > maxHeight {
				maxHeight = i
			}
			if j < minWidth {
				minWidth = j
			}
			if j > maxWidth {
				maxWidth = j
			}
		}

	}
	return (maxHeight - minHeigth + 1) * (maxWidth - minWidth + 1)
}
